engineering/repositories.md

```diff
@@ -30,6 +30,17 @@ Each repository should belong to a team, and the full team has admin rights over
 * A maintainer can merge his own PRs without code review, but it is encouraged to ask for a review  Merging without review, if done, should be reserved to fixing typos or minor maintenance tasks.
 * If a maintainer is missing, the Lead of the team that the project is owned by, or a backup maintainer designated by him will act as maintainer.
 
+## Commit messages
+
+Commit and Pull Request descriptions should be clear and detailed. Keep in mind that commit messages will be read by (eventually) a lot of people:
```

Comment:
> (eventually) a lot of people:

and processed by 🤖

Reply:
In my view, both are very important. Each commit in the PR should document the change made by that particular commit, and the PR overall should document why these commits are together.

Because the scope of individual commits and PRs is often quite different, the level of detail might vary. However, I would generally argue that they should have a similar structure:

1. A one line synopsis of what the commit/PR accomplishes

2. A more detailed explanation of why and what is being changed.

For tiny commits it may be OK to omit (2), particularly if it's updates from a code review. But even then there can be value—for example:

> Updates from code review.
>
> Fix spelling and punctuation errors, and reformat C files with clang-format.

I find the hardest thing about writing good, clear commit and PR descriptions is getting out of the present moment, where I know as much as I will ever know about this particular issue—and putting myself in the future, when someone else will have to reconstruct it. There is no simple formula for that, obviously—what I try to do is anticipate the questions someone is likely to ask me and answer them up front. YMMV.

Most (all?) of our projects seem to prefer _not_ squashing PRs when they are merged to master. I'd argue that it's therefore important that even our little code-review commits speak for themselves. In a repo where commits are expected to be squashed (even if not ahead-of-time), the PR description and the merge commit can basically be the same description.

Either way, though, the general advice proposed in this PR seems good.